src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -49,17 +49,15 @@ public PersistableSlidingWindow(int slidingWindowSize,
                                   Path filePath) {
     super(slidingWindowSize, timeUnit);
     this.pathToFile = filePath;
-    if (this.pathToFile == null) {
-      this.enablePersistence = false;
-    } else {
-      this.enablePersistence = true;
-      // setting the last write time to now will cause our first write to occur 5 minutes after construction
-      this.lastWriteTimeEpochMs = Instant.now().toEpochMilli();
-      try {
+    this.enablePersistence = this.pathToFile != null;
+    // setting the last write time to now will cause our first write to occur 5 minutes after construction
+    this.lastWriteTimeEpochMs = Instant.now().toEpochMilli();
+    try {
+      if (enablePersistence && Files.exists(filePath)) {
```

Comment:
Can we add a log message when enablePersistence is not set or when file does not exist in the path? Even debug level log would be great.

Reply:
Added more logging in the most recent commit